### PR TITLE
zola: unstable-2021-07-10 -> unstable-2021-07-14

### PR DESCRIPTION
**Pull request by D4ndellion**
###### Motivation for this change

Zola currently fails building: https://hydra.nixos.org/build/147776667
This bumps it and fixes the cargo sha hash to make it build again

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by legendofmiracles** [comment]
Sorry for breaking the hash, I'm positive that the last time i tested it worked - but oh well

**Comment by dali99** [comment]
@legendofmiracles don't worry about it, if it was wrong in your PR I assume it would've been caught by the reviewers :)

**Review by 0x4A6F — APPROVED** [review]
Lets bump this manually for now, because [0.14.0](https://github.com/getzola/zola/releases/tag/v0.14.0) fails to build:
```
   Compiling socket2 v0.4.0
error[E0658]: `match` is not allowed in a `const fn`
   --> /build/zola-0.14.0-vendor.tar.gz/socket2/src/lib.rs:156:9
    |
156 | /         match address {
157 | |             SocketAddr::V4(_) => Domain::IPV4,
158 | |             SocketAddr::V6(_) => Domain::IPV6,
159 | |         }
    | |_________^
    |
    = note: see issue #49146 <https://github.com/rust-lang/rust/issues/49146> for more information

error: aborting due to previous error

For more information about this error, try `rustc --explain E0658`.
error: could not compile `socket2`.
```

**Comment by dali99** [comment]
Only an issue until we bump rustPlatform up to the newest stable probaby

File: pkgs/applications/misc/zola/default.nix [hunk 1]
```diff
@@ -2,17 +2,17 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "zola";
-  version = "unstable-2021-07-10";
+  version = "unstable-2021-07-14";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/zola/default.nix:5]:
```suggestion
  version = "0.14.0";
```

File: pkgs/applications/misc/zola/default.nix [hunk 2]
```diff
@@ -2,17 +2,17 @@
...
 
   src = fetchFromGitHub {
     owner = "getzola";
     repo = pname;
     # unstable because the latest release fails to build
-    rev = "8c3ce7d7fbc0d585d4cbf27598ac7dfe5acd96f1";
-    sha256 = "sha256-Tw3u96ZPb0yUXvtJ+rna6nnb0a+KfTEiR/PPEadFxDA=";
+    rev = "312ffcb04c06c5f157b9fd2b944b858703238592";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/zola/default.nix:11]:
```suggestion
    rev = "v${version}";
```

**Pull request merged**